libs/storage/Tsavorite/cs/src/core/VarLen/SpanByte.cs

```diff
@@ -264,78 +257,42 @@ public SpanByte Deserialize()
         }
 
         /// <summary>
-        /// Reinterpret a fixed pointer as a serialized SpanByte
+        /// Reinterpret a fixed pointer as a serialized <see cref="SpanByte"/>
         /// </summary>
-        /// <param name="ptr"></param>
-        /// <returns></returns>
         public static ref SpanByte Reinterpret(byte* ptr)
         {
             return ref Unsafe.AsRef<SpanByte>(ptr);
         }
 
         /// <summary>
-        /// Reinterpret a fixed ref as a serialized SpanByte (user needs to write the payload length to the first 4 bytes)
+        /// Reinterpret a fixed ref as a serialized <see cref="SpanByte"/> (user needs to write the payload length to the first 4 bytes)
         /// </summary>
-        /// <returns></returns>
         public static ref SpanByte Reinterpret<T>(ref T t)
         {
             return ref Unsafe.As<T, SpanByte>(ref t);
         }
 
         /// <summary>
-        /// Create a SpanByte around a fixed Span&lt;byte&gt;. Warning: ensure the Span is fixed until operation returns.
+        /// Create a SpanByte around a pinned memory <paramref name="pointer"/> of given <paramref name="length"/>
         /// </summary>
-        /// <param name="span"></param>
-        /// <returns></returns>
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static SpanByte FromFixedSpan(Span<byte> span)
-        {
-            return new SpanByte(span.Length, (IntPtr)Unsafe.AsPointer(ref span[0]));
-        }
+        public static SpanByte FromPointer(byte* pointer, int length) => new(length, (nint)pointer);
 
         /// <summary>
-        /// Create a SpanByte around a fixed ReadOnlySpan&lt;byte&gt;. Warning: ensure the Span is fixed until operation returns.
+        /// Create a <see cref="SpanByte"/> from the given <paramref name="span"/>.
         /// </summary>
-        /// <param name="span"></param>
-        /// <returns></returns>
+        /// <remarks>
+        /// SAFETY: The <paramref name="span"/> MUST point to pinned memory.
+        /// </remarks>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static SpanByte FromFixedSpan(ReadOnlySpan<byte> span)
+        public static SpanByte FromPinnedSpan(ReadOnlySpan<byte> span)
         {
-            fixed (byte* ptr = span)
-            {
-                return new SpanByte(span.Length, (IntPtr)ptr);
-            }
+            return new SpanByte(span.Length, (nint)Unsafe.AsPointer(ref MemoryMarshal.GetReference(span)));
```

Comment:
I'm not sure it's wise to add another use of ``Unsafe.AsPointer``, which is just another place we'd need to think about GC holes

Reply:
In ideal world, I'd love to rip out of `SpanByte`, `SpanByteAndMemory` and `ArgSlice` out of existence and rewrite Tsavorite & Faster.. However, these changes are just imported from https://github.com/microsoft/FASTER/pull/903

Reply:
Yeah, I was on the fence whether to split it. I'll wait for maintainer opinion if that's okay for you? This should still be reviewable commit by commit, I think.

Reply:
This looks good. We used fixed here because `MemoryMarshal.GetReference(span)` wasn't available at the time.